### PR TITLE
Update raft api docs

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: website/content/api-docs/system/storage/raft.mdx [hunk 1]
```diff
@@ -39,6 +39,19 @@ leader node.
 - `leader_client_key` `(string: "")` - Client key used to communicate with
   Raft's leader node.
```
Comment (ncabatoff) [inline website/content/api-docs/system/storage/raft.mdx:41]:
```suggestion
- `leader_tls_servername` `(string: "")` - TLS servername to use when connecting with HTTPS.  Should match one of the names in the DNS SANs of the remote server certificate. 
```

**Review by ncabatoff — APPROVED** [review]
Looks good.  Mind squeezing in my addition while you're at it?

**Review by raskchanky — APPROVED** [review]

File: website/content/api-docs/system/storage/raft.mdx (hunk 1 above)
Comment (vishalnayak) [reply website/content/api-docs/system/storage/raft.mdx]: Sorry, I had missed out on that.

File: website/content/api-docs/system/storage/raft.mdx [hunk 2]
```diff
@@ -38,6 +38,19 @@ leader node.
...
+- `leader_tls_servername` `(string: "")` - TLS servername to use when connecting with HTTPS.  Should match one of the names in the DNS SANs of the remote server certificate. 
+- `auto_join` `(string: "")` - Defines any cloud auto-join metadata. If
+ supplied, Vault will attempt to automatically discover peers in addition to what
+ can be provided via 'leader_api_addr'.
+
+- `auto_join_scheme` `(string: "https")` - URI scheme to be used for `auto_join`.
+
+- `auto_join_port` `(int: 8200)` - Port to be used for `auto_join`.
+
+- `-non-voter` `(bool: false) (enterprise)` - If set, will make the server not
```
Comment (briankassouf) [inline website/content/api-docs/system/storage/raft.mdx:50]:
```suggestion
- `non-voter` `(bool: false) (enterprise)` - If set, will make the server not
```

File: website/content/api-docs/system/storage/raft.mdx [hunk 3]
```diff
@@ -38,6 +38,19 @@ leader node.
...
+- `leader_tls_servername` `(string: "")` - TLS servername to use when connecting with HTTPS.  Should match one of the names in the DNS SANs of the remote server certificate. 
+- `auto_join` `(string: "")` - Defines any cloud auto-join metadata. If
+ supplied, Vault will attempt to automatically discover peers in addition to what
+ can be provided via 'leader_api_addr'.
+
+- `auto_join_scheme` `(string: "https")` - URI scheme to be used for `auto_join`.
+
+- `auto_join_port` `(int: 8200)` - Port to be used for `auto_join`.
+
+- `non-voter` `(bool: false) (enterprise)` - If set, will make the server not
```
Comment (briankassouf) [inline website/content/api-docs/system/storage/raft.mdx:50]:
```suggestion
- `non_voter` `(bool: false) (enterprise)` - If set, will make the server not
```

I think this is `non_voter` https://github.com/hashicorp/vault-enterprise/blob/d6caa2452d2cb083ce0ad59b4f6539b1797ccad5/http/sys_raft.go#L112

**Pull request merged**